content/Hardware Support/UNO/Restore-the-connectivity-firmware-on-UNO-R4-WiFi-with-espflash.md

```diff
@@ -0,0 +1,62 @@
+---
+title: "Restore the connectivity firmware on UNO R4 WiFi with espflash"
+id: 16379769332892
+---
+
+The UNO R4 WiFi has a ESP32-S3 chip, which handles the USB-to-serial communication. The firmware for this chip can normally be upgraded [using Arduino IDE or Arduino Cloud](https://support.arduino.cc/hc/en-us/articles/9670986058780-Update-the-connectivity-module-firmware-on-UNO-R4-WiFi#iot), but if this is not possible the firmware can be restored using **espflash**.
+
+You may want to use this procedure if:
+
+* Your UNO R4 WiFi is detected as a generic ESP32 board by Arduino IDE or Arduino Cloud.
+* Your UNO R4 WiFi isn't detected as any board by Arduino IDE or Arduino Cloud, despite being connected to your computer with a working data USB cable.
+
+---
+
+Follow these steps:
+
+1. Unplug any non-essential USB devices from your computer.
+1. Short the pins highlighted in the image using a jumper wire:
+
+   ![The GND and Download ESP32 pins.](img/esp32-data-pins.png)
+1. Maintain the short between these pins as you connect the UNO R4 WiFi board to your computer with a USB cable.
+1. Download and extract the .ZIP file for your system:
+   * [unor4wifi-update-windows.zip](https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/latest/download/unor4wifi-update-windows.zip){.link-download}
+   * [unor4wifi-update-macos.zip](https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/latest/download/unor4wifi-update-macos.zip){.link-download}
+   * [unor4wifi-update-linux.zip](https://github.com/arduino/uno-r4-wifi-usb-bridge/releases/latest/download/unor4wifi-update-linux.zip){.link-download}
+1. Open your system's command line application inside the extracted folder.
+   * **Windows:** Hold <kbd>⇧Shift</kbd> and right-click any blank space inside the extracted folder. In the context menu, select **Open command window here / Open PowerShell window here**.
+   * **macOS:** Control-click on the unzipped `unor4wifi-update-macos` folder and select "**New Terminal at Folder**" from the context menu. A terminal window will open.
+   * **Linux:** Open [a command line terminal](https://ubuntu.com/tutorials/command-line-for-beginners) in the extracted folder.
+1. Run the command:
+   * **Windows:** `bin\espflash write-bin -b 115200 0x0 (Get-Item .\firmware\UNOR4-WIFI-S3-*.bin).FullName`
+   * **macOS/Linux:** `./bin/espflash write-bin -b 115200 0x0 firmware/UNOR4-WIFI-S3-*.bin`
```

Comment:
Good point. But since we're only using espflash in this procedure, allowing update.command to run shouldn't be neccessary. So we get

```suggestion
   * **macOS/Linux:** `./bin/espflash write-bin -b 115200 0x0 firmware/UNOR4-WIFI-S3-*.bin`
     > [!NOTE]
     > You may need to adjust your system's security settings to allow this command if you encounter a warning pop-up. To proceed, go to _System Preferences > Privacy & Security_, and under _Security_, click _Allow Anyway_ to allow espflash to run . 
   
```